proposals/records.md

```diff
@@ -1,75 +1,50 @@
-
 ## Members of a record type
 
-In addition to the members declared in the class or struct body, a record type has the following additional members:
-
-### Primary Constructor
-
-A record type has a public constructor whose signature corresponds to the value parameters of the
-type declaration. This is called the primary constructor for the type, and causes the implicitly
-declared default class constructor, if present, to be suppressed. It is an error to have a primary
-constructor and a constructor with the same signature already present in the class.
-
-At runtime the primary constructor
-
-1. executes the instance field initializers appearing in the class-body; and then
-    invokes the base class constructor with no arguments.
-
-1. initializes compiler-generated backing fields for the properties corresponding to the value parameters (if these properties are compiler-provided
-
-### Properties
-
-For each record parameter of a record type declaration there is a corresponding public property member whose name and type are taken from the value parameter declaration. If no concrete (i.e. non-abstract) property with a get accessor and with this name and type is explicitly declared or inherited, it is produced by the compiler as follows:
-
-For a record struct or a record class:
-
-* A public `get` and `init` auto-property is created (see separate `init` accessor specification). Its value is initialized during construction with the value of the corresponding primary constructor parameter. Each "matching" inherited abstract property's get accessor is overridden.
+In addition to the members declared in the record body, a record type has the following additional members:
 
 ### Equality members
 
-Record types produce synthesized implementations for the following methods:
+Record types produce synthesized implementations for the following methods, where `T` is the
+containing type:
 
-* `object.GetHashCode()` override, unless it is sealed or user provided
-* `object.Equals(object)` override, unless it is sealed or user provided
+* `object.GetHashCode()` override, unless it is sealed (error) or user provided
+* `object.Equals(object)` override, unless it is sealed (error) or user provided
 * `T Equals(T)` method, where `T` is the current type
+* `Type EqualityContract` get-only property
 
-`T Equals(T)` is specified to perform value equality such that `Equals` is
-true if and only if all the instance fields declared in the receiver type
-are equal to the fields of the other type.
+`EqualityContract` is a virtual instance property which returns `typeof(T)`. If it
+is present in the base type, the synthesized property overrides the base. If the base
+is sealed or non-virtual, an error is produced.
+
+`T Equals(T)` is specified to perform value equality such that `Equals` is true if and only if
+all the instance fields declared in the receiver type are equal to the fields of the other type,
```

Comment:
>declared [](start = 24, length = 8)

To be clear, this does not appear to handle fields in the base. That will be extremely confusing to users.